docs/howto/architectures.rst

```diff
@@ -0,0 +1,234 @@
+      run-on: arm64
+
+Building on ``amd64`` will produce one snap built for ``amd64``. Building on
+``arm64`` will produce one snap built for ``arm64``. Snapcraft will not raise
+an error when building on another architecture. Instead, it will ignore the
+``architectures`` keyword and build for the build-on architecture.
+
+If ``run-on`` is omitted, then it will assume the value of ``build-on``. The
+following snippet snippet will produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+    - build-on: arm64
+
+The shorthand format will also produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - amd64
+    - arm64
+
+How to create an architecture independent snap
+----------------------------------------------
+
+core22
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: all
+
+core20
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      run-on: all
+
+How to create a cross-compiling snap
+------------------------------------
+
+core22
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: arm64
+
+Building on ``amd64`` will produce one snap built for ``arm64``. Snapcraft will
+raise an error when building on another architecture.
+
+``core22`` can handle complex build plans. For example:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: amd64
+    - build-on: [amd64, arm64]
+      build-for: arm64
+
+Building on ``amd64`` will produce two snaps, one built for ``amd64`` and one
+built for ``arm64``. Building on ``arm64`` will produce one snap built for
+``arm64``. Snapcraft will raise an error when building on another architecture.
+
+core20
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      run-on: arm64
+
+Building on ``amd64`` will produce one snap built for ``arm64``. Snapcraft will
+not raise an error when building on another architecture. Instead, it will
+ignore the ``architectures`` keyword and build for the build-on architecture.
+
+Complex build plans like the previous ``core22`` example are not supported for
+``core20``.
+
+How to stage packages from another architecture
+-----------------------------------------------
+
+To use an ``i386`` package for an ``amd64`` snap, use the following
+``snapcraft.yaml`` snippets for ``core22``:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+
+  package-repositories:
+    - type: apt
+      formats: [deb]
+      architectures: [i386]
+      components: [main]
+      suites: [jammy]
+      key-id: F23C5A6CF475977595C89F51BA6932366A755776
+      url: https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu
+
+  parts:
+    mypart:
+      stage-packages:
+        - libpython3.11-minimal:i386
+
+This is supported for related architectures. ``amd64`` and ``i386`` can stage
+packages for each other. ``arm64`` and ``armhf`` can also stage packages for
```

Comment:
`arm64` can stage packages for `armhf`, but can `armhf` stage packages for `arm64`? (Same question technically applies to amd64/i386 too, but is moot since I don't think you can actually build on `i386` at this point)

Reply:
It is bidirectional.  You could build on your own `i386` hardware and stage an `amd64` package.

I've made it more clear in the docs.